apis/v1beta1/validation/gateway.go

```diff
@@ -42,6 +44,9 @@ var (
 		gatewayv1b1.HTTPSProtocolType: {},
 		gatewayv1b1.TLSProtocolType:   {},
 	}
+
+	validHostnameAddress   = `^(\*\.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+	validHostnameRegexp, _ = regexp.Compile(validHostnameAddress)
```

Comment:
could `regexp.MustCompile()` be better?

Reply:
Yes, if you have a constant/literal string holding your regex that will ALWAYS compile the same way, use `MustCompile`.

Reply:
great, thanks